v2/inc/c_logging/log_context.h

```diff
@@ -4,6 +4,241 @@
+    (void)(void (*)(int MU_FOR_EACH_1(DEFINE_CONTEXT_NAME_AS_PARAMETER, __VA_ARGS__)))0x4242; \
+
+// SETUP_PROPERTY_PAIR
+
+#define EXPAND_SETUP_PROPERTY_PAIR_LOG_CONTEXT_STRING_PROPERTY(property_name, ...) \
+    /* Codes_SRS_LOG_CONTEXT_01_007: [ LOG_CONTEXT_STRING_PROPERTY shall expand to code allocating a property/value pair of type ascii_char_ptr and the name property_name. ]*/ \
+    property_value_pair->value = data_pos; \
+    property_value_pair->name = MU_TOSTRING(property_name); \
+    property_value_pair->type = &ascii_char_ptr##_log_context_property_type; \
+    /* Codes_SRS_LOG_CONTEXT_01_008: [ LOG_CONTEXT_STRING_PROPERTY shall expand to code that stores as value a string that is constructed using printf-like formatting based on format and all the arguments in .... ]*/ \
+    data_pos += sprintf(property_value_pair->value, __VA_ARGS__) + 1; \
+    property_value_pair++; \
+
+#define EXPAND_SETUP_PROPERTY_PAIR_LOG_CONTEXT_NAME(log_context_name) \
+    /* Codes_SRS_LOG_CONTEXT_01_012: [ The name of the struct property shall be the context name specified by using LOG_CONTEXT_NAME (if specified). ]*/ \
+    /* Codes_SRS_LOG_CONTEXT_01_017: [ The name of the struct property shall be the context name specified by using LOG_CONTEXT_NAME (if specified). ]*/ \
+    first_property_value_pair->name = MU_TOSTRING(log_context_name); \
+
+#define EXPAND_SETUP_PROPERTY_PAIR_LOG_CONTEXT_PROPERTY(property_type, property_name, field_value) \
+    /* Codes_SRS_LOG_CONTEXT_01_004: [ LOG_CONTEXT_PROPERTY shall expand to code allocating a property/value pair entry with the type property_type and the name property_name. ]*/ \
+    property_value_pair->value = data_pos; \
+    property_value_pair->name = MU_TOSTRING(property_name); \
+    property_value_pair->type = &property_type##_log_context_property_type; \
+    /* Codes_SRS_LOG_CONTEXT_01_005: [ LOG_CONTEXT_PROPERTY shall expand to code copying the value property_value to be the value of the property/value pair. ]*/ \
+    (void)property_type##_log_context_property_type_init((void*)data_pos, field_value); \
+    data_pos += sizeof(property_type); \
+    property_value_pair++; \
+
+#define SETUP_PROPERTY_PAIR(field_desc) \
+    MU_C2(EXPAND_SETUP_PROPERTY_PAIR_, field_desc)
+
+// COUNT_PROPERTY
+
+#define EXPAND_COUNT_PROPERTY_LOG_CONTEXT_STRING_PROPERTY(property_name, ...) \
+    + 1
+
+#define EXPAND_COUNT_PROPERTY_LOG_CONTEXT_NAME(log_context_name) \
+
+#define EXPAND_COUNT_PROPERTY_LOG_CONTEXT_PROPERTY(property_type, property_name, field_value) \
+    + 1
+
+#define COUNT_PROPERTY(field_desc) \
+    MU_C2(EXPAND_COUNT_PROPERTY_, field_desc)
+
+// COUNT_DATA_BYTES
+
+#define EXPAND_COUNT_DATA_BYTES_LOG_CONTEXT_STRING_PROPERTY(property_name, ...) \
+    + snprintf(NULL, 0, __VA_ARGS__) + 1
+
+#define EXPAND_COUNT_DATA_BYTES_LOG_CONTEXT_NAME(log_context_name) \
+
+#define EXPAND_COUNT_DATA_BYTES_LOG_CONTEXT_PROPERTY(property_type, property_name, field_value) \
+    + sizeof(property_type)
```

Comment:
>sizeof(property_type) [](http://example.com/codeflow?start=6&length=21)

LOG_CONTEXT_PROPERTY_TYPE_GET_INIT_DATA_SIZE I guess it is not useable in this context...

Reply:
But it should be... if snprintf is...

Reply:
Changed